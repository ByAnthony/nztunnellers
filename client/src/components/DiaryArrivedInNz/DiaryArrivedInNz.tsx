import STYLES from '../ProfileDiary/ProfileDiary.module.scss';

type Props = {
    inNzLength: string | null,
}

export function DiaryArrivedInNz({ inNzLength }: Props) {
  const displayImmigrationYear = (immigrationYear: string) => (
    <div className={STYLES['fullwidth-main-card']}>
      <p>Arrive in New Zealand</p>
      <span>{ immigrationYear }</span>
    </div>
  );

  return (
    <div>
      { inNzLength ? displayImmigrationYear(inNzLength) : null }
    </div>
  );
}
