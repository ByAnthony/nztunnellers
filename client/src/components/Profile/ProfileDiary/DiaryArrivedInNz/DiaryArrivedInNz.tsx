import STYLES from '../ProfileDiary.module.scss';

type Props = {
    inNzLength: string | null;
}

export function DiaryArrivedInNz({ inNzLength }: Props) {
  return (
    inNzLength ? (
      <div className={STYLES['fullwidth-main-card']}>
        <p>Settled in New Zealand</p>
        <span>{ inNzLength }</span>
      </div>
    ) : null
  );
}
