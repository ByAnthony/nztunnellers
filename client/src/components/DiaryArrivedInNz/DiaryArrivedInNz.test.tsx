import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { DiaryArrivedInNz } from './DiaryArrivedInNz';

const component = (
  <DiaryArrivedInNz inNzLength="1903" />
);

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});

test('renders the component when date known', () => {
  render(component);

  expect(screen.getByText('Settled in New Zealand')).toBeInTheDocument();
  expect(screen.getByText('1903')).toBeInTheDocument();
});

test('does not render the component when date unknown', () => {
  const { container } = render(
    <DiaryArrivedInNz inNzLength={null} />,
  );

  expect(container).toBeEmptyDOMElement();
});
