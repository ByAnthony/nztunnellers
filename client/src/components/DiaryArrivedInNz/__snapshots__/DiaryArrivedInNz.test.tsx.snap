// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="fullwidth-main-card"
  >
    <p>
      Settled in New Zealand
    </p>
    <span>
      1903
    </span>
  </div>
</DocumentFragment>
`;
